### PR TITLE
Implemented Main/View page

### PR DESCRIPTION
**Pull request by amDavid01**

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
+      "isBackendOnly": false,
+      "order": 30
     }
   },
   "proxy": {
     "whitelist": [
       {
-        "url": "https://jsonplaceholder.typicode.com/.*",
-        "methods": ["GET"],
+        "url": "https://(.*)/wp-json/wc/v3/.*",
```
Comment (zpawn) [inline manifest.json]: We can also use a placeholder in the manifest (example in [gitlab app](https://github.com/DeskproApps/gitlab/blob/master/manifest.json#L83))

in your case it'll be like this:
```json
"whitelist": [{
    "url": "__store_url__/wp-json/wc/v3/.*"
}]

File: manifest.json [hunk 2]
```diff
@@ -1,28 +1,52 @@
...
+    },
+    "consumer_secret": {
+      "title": "Consumer Secret",
+      "description": "Consumer secret can be obtained by following our setup guide",
+      "type": "string",
+      "isRequired": true,
+      "isBackendOnly": true,
+      "order": 20
+    },
+    "store_url": {
```
Comment (zpawn) [inline manifest.json:32]: Add validation url please for better UX.

example: https://github.com/DeskproApps/gitlab/blob/master/manifest.json#LL50C19-L50C19

File: src/pages/Main.tsx [hunk 3]
```diff
@@ -1,79 +1,108 @@
...
-  // Add a "refresh" button @see https://support.deskpro.com/en-US/guides/developers/app-elements
+  const { context } = useDeskproLatestAppContext();
+  const navigate = useNavigate();
   useInitialisedDeskproAppClient((client) => {
-    client.registerElement("myRefreshButton", { type: "refresh_button" });
+    client.setTitle("WooCommerce");
+
+    client.registerElement("refreshButton", { type: "refresh_button" });
+
+    client.registerElement("xeroHomeButton", {
```
Comment (zpawn) [inline src/pages/Main.tsx]: typo

File: src/pages/Main.tsx [hunk 4]
```diff
@@ -1,79 +1,108 @@
...
+    client.deregisterElement("link");
   });
 
-  // Listen for the "change" event and store the context data
-  // as local state @see https://support.deskpro.com/en-US/guides/developers/app-events
   useDeskproAppEvents({
-    onChange: setTicketContext,
+    async onElementEvent(id) {
+      switch (id) {
+        case "xeroHomeButton":
```
Comment (zpawn) [inline src/pages/Main.tsx]: typo

File: src/hooks/useQueryWithClient.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import { IDeskproClient, useDeskproAppClient } from "@deskpro/app-sdk";
+import {
+  QueryKey,
+  useMutation,
+  useQuery,
+  UseQueryOptions,
+  UseQueryResult,
+} from "@tanstack/react-query";
+
+export const useQueryWithClient = <
```
Comment (zpawn) [inline src/hooks/useQueryWithClient.ts]: we have this hook in the `@deskpro/app-sdk` (redundant code increases the bundle size)

File: src/pages/View/View.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+          : `wp-admin/user-edit.php?user_id=${id}`
+      }`,
+      hasIcon: true,
+    });
+  });
+
+  useDeskproAppEvents({
+    async onElementEvent(id) {
+      switch (id) {
+        case "xeroHomeButton":
```
Comment (zpawn) [inline src/pages/View/View.tsx]: type

File: src/pages/View/View.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        isOrder
+          ? `wp-admin/post.php?post=${id}&action=edit`
+          : `wp-admin/user-edit.php?user_id=${id}`
+      }`,
+      hasIcon: true,
+    });
+  });
+
+  useDeskproAppEvents({
+    async onElementEvent(id) {
```
Comment (zpawn) [inline src/pages/View/View.tsx]: why do you have `async` here?

File: src/components/Notes/Notes.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        >
+          <Stack gap={15}>
+            <P11 style={{ color: theme.colors.grey80 }}>
+              {formatDateSince(new Date(note.date_created))}
+            </P11>
+            <P1>{note.note}</P1>
+          </Stack>
+          <HorizontalDivider
+            style={{
+              width: "115%",
```
Comment (zpawn) [inline src/components/Notes/Notes.tsx]: In the `HorizontalDivider` component, you specify the width to 110%, but here 115% is a potential **bug** with the appearance of a horizontal scroll

Comment (zpawn) [reply src/components/Notes/Notes.tsx]: Let's add `<body style="margin: 0">` in the `index.html`, and for `margin` create a container in which you will wrap the blocks. And the `HorizontalDivider ` component will have a width of 100%, which excludes horizontal scrolling

File: src/hooks/useQueryWithClient.ts (hunk 5 above)
Comment (iTweeno) [reply src/hooks/useQueryWithClient.ts:11]: Not wrapped in the react query context, so won't work, need to fix that

File: src/components/Notes/Notes.tsx (hunk 8 above)
Comment (amDavid01) [reply src/components/Notes/Notes.tsx]: replaced for 100vw, and as I said in all other apps there is no horizontal scrolling

**Review by zpawn — APPROVED** [review]

**Review by AshleyDawson — APPROVED** [review]

**Pull request merged**